README.md

```diff
@@ -76,3 +76,25 @@ The framework serves two distinct, but related use cases:
 - [Command line interface](docs/CLI.md)
 - [Contribution guide](docs/CONTRIBUTING.md)
 - [Upgrading](docs/UPGRADING.md)
+
+### Version compatibility with Elasticsearch
+
+> [!NOTE]
+> Version compatibility will not be checked if Elasticsearch is serverless.
+
+The Connector will perform a version compatibility check with the configured Elasticsearch server on startup.
+If the versions are incompatible, the Connector will terminate and output the incompatible versions in the shell.
+If the versions are different but otherwise compatible, the Connector will output a warning in the shell but will continue operating.
+
+We recommend running on the same version as Elasticsearch.
+However, if you want to hold back upgrading one or the other for any reason, use this table to determine if your versions will be compatible.
+
+| Situation                 | Example Conenctor Framework version | Example ES version | Outcome |
+| ------------------------- |-------------------------------------|--------------------| ------- |
+| Versions are the same.    | 8.15.1                              | 8.15.1             | 💚 OK      |
```

Comment:
```suggestion
| Versions are the same.    | 8.15.1.0                              | 8.15.1             | 💚 OK      |
```

Doesn't really matter, but might be good to include BUILD versions in these examples

Reply:
Added, also added a row specifying that build version doesn't matter.